analyzers/src/SonarAnalyzer.CSharp/Rules/UseModelBinding.cs

```diff
@@ -0,0 +1,253 @@
+            return true;
+        }
+        // Within a single code block, access via constant and variable keys could be mixed
+        // We only want to raise, if all access were done via constants
+        var allConstantAccess = true;
+        var codeBlockCandidates = new ConcurrentStack<ReportCandidate>();
+        if (argumentDescriptors.Any())
+        {
+            codeBlockStart.RegisterNodeAction(nodeContext =>
+            {
+                var argument = (ArgumentSyntax)nodeContext.Node;
+                var context = new ArgumentContext(argument, nodeContext.SemanticModel);
+                if (allConstantAccess && Array.Exists(argumentDescriptors, x => Language.Tracker.Argument.MatchArgument(x)(context)))
+                {
+                    allConstantAccess &= nodeContext.SemanticModel.GetConstantValue(argument.Expression) is { HasValue: true, Value: string };
+                    codeBlockCandidates.Push(new(UseModelBindingMessage, GetPrimaryLocation(argument), OriginatesFromParameter(nodeContext.SemanticModel, argument)));
+                }
+            }, SyntaxKind.Argument);
+        }
+        if (propertyAccessDescriptors.Any())
+        {
+            codeBlockStart.RegisterNodeAction(nodeContext =>
+            {
+                var memberAccess = (MemberAccessExpressionSyntax)nodeContext.Node;
+                var context = new PropertyAccessContext(memberAccess, nodeContext.SemanticModel, memberAccess.Name.Identifier.ValueText);
+                if (Language.Tracker.PropertyAccess.MatchProperty(propertyAccessDescriptors)(context))
+                {
+                    codeBlockCandidates.Push(new(UseIFormFileBindingMessage, memberAccess.GetLocation(), OriginatesFromParameter(nodeContext.SemanticModel, memberAccess)));
+                }
+            }, SyntaxKind.SimpleMemberAccessExpression);
+        }
+        codeBlockStart.RegisterCodeBlockEndAction(codeBlockEnd =>
+        {
+            if (allConstantAccess)
+            {
+                controllerCandidates.PushRange([.. codeBlockCandidates]);
+            }
+        });
+        return false;
+    }
+
+    private static (ArgumentDescriptor[] ArgumentDescriptors, MemberDescriptor[] PropertyAccessDescriptors) GetDescriptors(Compilation compilation)
+    {
+        var argumentDescriptors = new List<ArgumentDescriptor>();
+        var propertyAccessDescriptors = new List<MemberDescriptor>();
+        if (compilation.GetTypeByMetadataName(KnownType.Microsoft_AspNetCore_Mvc_ControllerAttribute) is { })
+        {
+            AddAspNetCoreDescriptors(argumentDescriptors, propertyAccessDescriptors);
+        }
+        // TODO: Add descriptors for Asp.Net MVC 4.x
+        return ([.. argumentDescriptors], [.. propertyAccessDescriptors]);
+    }
+
+    private static void AddAspNetCoreDescriptors(List<ArgumentDescriptor> argumentDescriptors, List<MemberDescriptor> propertyAccessDescriptors)
```

Comment:
From my point of view, the ArgumentDescriptor should be moved to a "KnownArguments" class and be static properties there, pretty much like "KnownMethods" and "KnownTypes". Ideally the "Known..." approach and the "Tracker..." approach are combined into a single one because they are two mechanisms that do the same.
We need a dedicated effort to achieve that.

I went for the list-based populating approach to be more flexible. e.g. if https://github.com/dotnet/aspnetcore/issues/7538 is implemented, we could at Cookies to the collection depending on a version check.